src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -93,5 +94,56 @@ public virtual bool CanParseValue(object value)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Checks that the provided value is a valid default value.
+        /// This method should not throw an exception.
+        /// </summary>
+        public virtual bool IsValidDefault(object value)
+        {
+            try
+            {
+                return ToAst(value) != null;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Converts a value to an AST representation. This is necessary for introspection queries
+        /// to return the default value for fields of this scalar type. This method may throw an exception
+        /// or return null for a failed conversion.
+        /// </summary>
+        public virtual IValue ToAst(object value)
+        {
+            try
+            {
+                var serialized = Serialize(value);
+                return serialized switch
+                {
+                    bool b => new BooleanValue(b),
+                    byte b => new IntValue(b),
+                    sbyte sb => new IntValue(sb),
+                    short s => new IntValue(s),
+                    ushort us => new IntValue(us),
+                    int i => new IntValue(i),
+                    uint ui => new LongValue(ui),
+                    long l => new LongValue(l),
+                    ulong ul => new BigIntValue(ul),
+                    BigInteger bi => new BigIntValue(bi),
+                    decimal d => new DecimalValue(d),
+                    float f => new FloatValue(f),
+                    double d => new FloatValue(d),
+                    string s => new StringValue(s),
+                    _ => new StringValue(value.ToString())
```

Comment:
NRE here in case of `null` from `Serialize`. Add `null => null` before it.

Reply:
Hmmm... if `Serialize` returns null, it means that the value failed to convert.  And `Serialize(null)` should never be called.  So perhaps `ToAst(null)` should `return new NullValue()`, and throw an exception if `Serialize` returns `null`?

Reply:
```suggestion
                    null => null,
                    _ => new StringValue(value.ToString())
```

Reply:
I see that
```
  // Since value is an internally represented value, it must be serialized
            // to an externally represented value before converting into an AST.
            var serialized = inputType.Serialize(value) ?? throw new InvalidOperationException($"Unable to serialize '{value}' to '{inputType.Name}'.");
``` 
before.

Reply:
Yes although the new `ToAst` method is defined as allowing `null` response in case of a failure, same as `Serialize` does.  See GraphQLExtensions around line 595:

```cs
            // Populate the fields of the input object by creating ASTs from each value
            // in the dictionary according to the fields in the input type.
            if (type is IInputObjectGraphType input)
            {
                return input.ToAst(value) ?? throw new InvalidOperationException($"Unable to convert the '{value}' of the object type '{input.Name}' to an AST representation.");
            }

            if (!(type is ScalarGraphType scalar))
                throw new ArgumentOutOfRangeException(nameof(type), $"Must provide Input Type, cannot use: {type}");

            return scalar.ToAst(value) ?? throw new InvalidOperationException($"Unable to convert '{value}' of the scalar type '{scalar.Name}' to an AST representation.");
```